Handle python 3.11 for huggingface["video"] and dependencies

### Feature request

Currently, the only thing preventing the installation of:
- huggingface[dev]
- huggingface[all]
- huggingface[video]

On python 3.11 is the `av==9.2.0` requirement. That requirement cites issue with `av==10.0.0` as reason for pinning this version, however the latest version is now `14.0.1`, it would be nice to revisit that pin, especially now that it forces users on python 3.10.

### Motivation

It would allow contributors and users of huggingface[video] to use python 3.11.

### Your contribution

I've looked into the issue with av==9.2.0, the original failure is Cython generating bad C:
```
      av/logging.pyx:351:28: Cannot assign type 'void (void *, int, const char *, va_list) except * nogil' to 'av_log_callback' (alias of 'void (*)(void *, int, const char *, va_list) noexcept nogil'). Exception values are incompatible. Suggest adding 'noexcept' to the type of 'log_callback'.
```

This can be circumvented by forcing an earlier version of Cython:
```
$ echo "cython<3.0" >> c.txt
$ PIP_CONSTRAINT=c.txt pip install av==9.2.0
```
But that leads to this error:
`src/av/stream.c:2999:29: error: ‘struct AVStream’ has no member named ‘codec’`
Which, looking into it, means that ffmpeg is too recent for this av code, that's where I give up because I realised that asking users to downgrade ffmpeg was probably not the way to go.

I'm attempting to run the tests on python 3.10 w/ av==9.2.0 vs python 3.11 w/ av==14.0.1, but `make test` takes absolute ages even on 32 cores and 64Gb RAM and it just seems to fail randomly so clearly I am under equiped for this undertaking. Any tips to have a better times with running tests? Is there a way to only run those using `av` besides `grep "import av"`?

--- a/setup.py
+++ b/setup.py
@@ -97,7 +97,7 @@ if stale_egg_info.exists():
 _deps = [
     "Pillow>=10.0.1,<=15.0",
     "accelerate>=0.26.0",
-    "av==9.2.0",  # Latest version of PyAV (10.0.0) has issues with audio stream.
+    "av",
     "beautifulsoup4",
     "blobfile",
     "codecarbon>=2.8.1",

--- a/src/transformers/dependency_versions_table.py
+++ b/src/transformers/dependency_versions_table.py
@@ -4,7 +4,7 @@
 deps = {
     "Pillow": "Pillow>=10.0.1,<=15.0",
     "accelerate": "accelerate>=0.26.0",
-    "av": "av==9.2.0",
+    "av": "av",
     "beautifulsoup4": "beautifulsoup4",
     "blobfile": "blobfile",
     "codecarbon": "codecarbon>=2.8.1",

